qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDefinition.java

```diff
@@ -378,35 +376,36 @@ private CompletableFuture<List<? extends LocationLink>> findDefinitionFromParame
 	}
 
 	private CompletableFuture<List<? extends LocationLink>> findDefinitionFromParameterDataModel(Part part,
-			Template template, String projectUri, CancelChecker cancelChecker) {
+			Template template, QuteProject project, CancelChecker cancelChecker) {
 		return template.getParameterDataModel(part.getPartName()) //
 				.thenCompose(parameter -> {
 					cancelChecker.checkCanceled();
 					// Try to find in parameter model
 					if (parameter != null) {
-						QuteJavaDefinitionParams params = parameter.toJavaDefinitionParams(projectUri);
-						return findJavaDefinition(params, cancelChecker, () -> QutePositionUtility.createRange(part));
+						QuteJavaDefinitionParams params = parameter.toJavaDefinitionParams(project.getUri());

```

Comment:
parameter.toJavaDefinitionParams is used too for codelens which is working with project uri. Generally for API it is better to set the minimal information (in this case String instead of QueProject) in order to use it in any context (with or without QuteProject).

Reply:
Okay. That makes sense